### PR TITLE
Create server.js

### PR DESCRIPTION
**Pull request by sjuhi**

### CODE REVIEW

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Assigning onNewHireCreateCallback to onNewHireCreate event
+ */
+  onNewHireCreateCallback: function(payload) {
+    console.log('payload.iparams.apiKey')
+	console.log(payload.iparams.apiKey)
+	 sgMail.setApiKey(payload.iparams.apiKey);
+        const msg = {
+          to: payload.data.newhire.user_emails[0],
+          from: 'sjuhi1818@gmail.com',
```
Comment (hemchander23) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: This should be set via iparams. Avoid hardcoding stuff

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+exports = {
+  events: [
+    { event: "onNewHireCreate", callback: "onNewHireCreateCallback"},
+  ],
+/**
+ * Assigning onNewHireCreateCallback to onNewHireCreate event
+ */
+  onNewHireCreateCallback: function(payload) {
+    console.log('payload.iparams.apiKey')
+	console.log(payload.iparams.apiKey)
```
Comment (hemchander23) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: Kindly avoid logging iparams. Esp key. Only log the errors.

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	 sgMail.setApiKey(payload.iparams.apiKey);
+        const msg = {
+          to: payload.data.newhire.user_emails[0],
+          from: 'sjuhi1818@gmail.com',   
+          subject: 'Feedback Form for New Hires',
```
Comment (hemchander23) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: Subject, text, html should be set via iparams. 
Add a function to sanitize html before sending it in the payload

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          text: '',
+          html: "Hello! Welcome to Freshworks.It's been a month as you have joined the company.We were happy to have you onboard.Hope your experience with us so far has been great. Your feedback on our hiring and on-boarding processes is of paramount importance to better ourselves.We hope you'll help us by taking the survey.Please make some time to fill out the feedback form attached with this email. Hope this suffices",
+        };
+        sgMail.send(msg);
```
Comment (hemchander23) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: How is the error handled here?

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+
+/**
+ * Sendgrid API is used to Send mails 
+ */
+const sgMail = require('@sendgrid/mail');
+const htmlMsg = require('./Template').html_msg;
+exports = {
+  events: [
+    { event: "onNewHireCreate", callback: "onNewHireCreateCallback"},
```
Comment (hemchander23) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: The code creates a scheduled event. However, I don't find a handler for it

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+
+/**
+ * Sendgrid API is used to Send mails 
+ */
+const sgMail = require('@sendgrid/mail');
+const htmlMsg = require('./Template').html_msg;
```
Comment (hemchander23) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: The template uses certain hardcoded URLs. Survey form and text inside the template should be substituted via iparams. Kindly refactor https://github.com/freshdesk/marketplace-sample-apps/blob/master/Freshworks-Samples/Freshteam/newhire-feedback-collector/server/Template.js#L407

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+exports = {
+  events: [
+    { event: "onNewHireCreate", callback: "onNewHireCreateCallback"},
+  ],
+/**
+ * Assigning onNewHireCreateCallback to onNewHireCreate event
+ */	
+     onNewHireCreateCallback: function(payload) {
+	console.log(htmlMsg);
+    console.log('payload.iparams.apiKey')
```
Comment (hemchander23) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: Beautify the code

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        };
+        sgMail.send(msg);
+/**
+ *Using Shedule event for sending mail after 30 days
+ */
+    let d = new Date();
+    d.setMonth(d.getMonth()+1);
+    $schedule.create({
+      name: "ticket_reminder",
+      data: {ticket_id: 100001},
```
Comment (hemchander23) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: Ideally, this should be the data that sits inside the template. Kindly refactor

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  events: [
+    { event: "onNewHireCreate", callback: "onNewHireCreateCallback"},
+  ],
+/**
+ * Assigning onNewHireCreateCallback to onNewHireCreate event
+ */	
+     onNewHireCreateCallback: function(payload) {
+	console.log(htmlMsg);
+    console.log('payload.iparams.apiKey')
+	console.log(payload.iparams.apiKey)
```
Comment (raghu12133) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: console logs can be removed.

File: Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    let d = new Date();
+    d.setMonth(d.getMonth()+1);
+    $schedule.create({
+      name: "ticket_reminder",
+      data: {ticket_id: 100001},
+      schedule_at: d.toISOString(),
+    })
+  }
+}
+
```
Comment (raghu12133) [inline Freshworks-Samples/Freshteam/newhire-feedback-collector/server.js]: Empty new lines can be removed.

**Pull request closed without merging**